admin/controllers/tech-support.js

```diff
@@ -507,6 +508,50 @@ const controller = {
     } catch (error) {
       return next(error)
     }
+  },
+
+  getPsReportFolders: async function getPsReportFolders (req, res, next) {
+    try {
+      const logs = await psReportLogsDownloadService.getLogFoldersList()
+      res.locals.pageTitle = 'PS Report Logs'
+      req.breadcrumbs('PS Report Logs')
+      res.render('tech-support/ps-report-logs', {
+        breadcrumbs: req.breadcrumbs(),
+        logs: logs
+      })
+    } catch (error) {
+      return next(error)
+    }
+  },
+
+  getPsReportFolderFileList: async function getPsReportFolderFileList (req, res, next) {
+    try {
+      const files = await psReportLogsDownloadService.getLogFolderFileList(req.params.folder)
+      res.locals.pageTitle = 'PS Report Log Folder Files'
+      req.breadcrumbs('PS Report Logs', '/tech-support/ps-report-logs')
+      req.breadcrumbs('PS Report Log Folder Files')
+      res.render('tech-support/ps-report-log-folder', {
+        breadcrumbs: req.breadcrumbs(),
+        files: files,
+        folder: req.params.folder
+      })
+    } catch (error) {
+      return next(error)
+    }
+  },
+
+  getPsReportLogFileContents: async function getPsReportLogFileContents (req, res, next) {
+    try {
+      const file = await psReportLogsDownloadService.downloadLogFile(req.params.folder, req.params.file)
+      res.set({
+        'Content-Disposition': `attachment; filename="${req.params.file}"`,
```

Comment:
Are there any security implications in trusting the Request parameter here?

Reply:
good point.  these need sanitising.  will add some validation and narrowing checks